docs/plainmonthday.md

```diff
@@ -20,18 +20,22 @@ A `Temporal.PlainMonthDay` can be converted into a `Temporal.PlainDate` by combi
 - `isoMonth` (number): A month, ranging between 1 and 12 inclusive.
 - `isoDay` (number): A day of the month, ranging between 1 and 31 inclusive.
 - `calendar` (optional `Temporal.Calendar` or plain object): A calendar to project the date into.
-- `referenceISOYear` (optional number): A reference year, used for disambiguation when implementing other calendar systems.
-  The default is the first leap year after the [Unix epoch](https://en.wikipedia.org/wiki/Unix_time).
-  You can omit this parameter unless using a non-ISO-8601 calendar.
+- `referenceISOYear` (optional for ISO 8601 calendar; required for other calendars):
+  A reference year in the ISO 8601 calendar for disambiguation when implementing calendar systems.
+  The default for the ISO 8601 calendar is the first leap year after the [Unix epoch](https://en.wikipedia.org/wiki/Unix_time).
+  This value is required for other calendar systems, but can cause serious problems if not set correctly.
 
 **Returns:** a new `Temporal.PlainMonthDay` object.
 
-Use this constructor if you have the correct parameters for the date already as individual number values, or you are implementing a custom calendar.
-Otherwise, `Temporal.PlainMonthDay.from()`, which accepts more kinds of input, allows inputting dates in different calendar reckonings, and allows controlling the overflow behaviour, is probably more convenient.
+> NOTE: To avoid infinite recursion, the `referenceISOYear` is accepted as-is without validating that the year contains the specified month and day in the desired calendar and without replacing the year with a canonical value like `from` and `with` will do.
+> These limitations mean that `equals` or `compare` may return `false` for `Temporal.PlainMonthDay` instances where the month and day are identical, but the reference years don't match.
+> For this reason, it is STRONGLY recommended that this constructor SHOULD NOT be used except when implementing a custom calendar or when only using the ISO 8601 calendar.
+> For other calendars, use the `from` method which will always set the same reference year for the same month, day, and calendar.
```

Comment:
I'd probably find this a bit too much detail if I just want to use the PlainMonthDay type. How about this?
```suggestion
> The `calendar` and `referenceISOYear` parameters should only be used in specialized cases such as custom calendars, because `equals` or `compare` will consider these two objects unequal: `new Temporal.PlainMonthDay(3, 14, 'iso8601', 1970)` and `new Temporal.PlainMonthDay(3, 14, 'iso8601`, 2000)`.
> Normally, it is _strongly_ recommended to use the two-argument form or the `from()` method.
```

Reply:
I made some tweaks to the wording, but otherwise I like your edits!

```suggestion
> The `calendar` and `referenceISOYear` parameters should be avoided because `equals` or `compare` will consider these two objects unequal: `new Temporal.PlainMonthDay(3, 14, 'iso8601', 1970)` and `new Temporal.PlainMonthDay(3, 14, 'iso8601`, 2000)`.
> When creating instances for non-ISO-8601 calendars (other than when implementing a custom calendar) use the `from()` method which will automatically set a valid and consistent reference year for all calendars.
```